File: programming/6.1/main.py
from random import randint


def genearate_payment(data):
  for i in range(len(data)):
    data[i] = randint(10000, 15000) / 100 * 80

  return data

def generate_time(data):
  for i in range(len(data)):
    data[i] = randint(0, 32)

  return data

y_time = [i for i in range(10)]
g_time = [i for i in range(15)]

y_payment = [i for i in range(10)]
g_payment = [i for i in range(15)]

y_time = generate_time(y_time)
g_time = generate_time(g_time)

y_payment = genearate_payment(y_payment)
g_payment = genearate_payment(g_payment)

y_average_payment = 0
g_average_payment = 0

for i in y_payment:
  y_average_payment += i

for i in g_payment:
  g_average_payment += i

y_average_payment = y_average_payment / 10
g_average_payment = g_average_payment / 15

if y_average_payment > g_average_payment:
  print('Y avarage payment time more than G')

else:
  print('G avarage payment time more than Y')

File: programming/6.1/main.cpp
#include <iostream>
#include <cmath>
#include <ctime>
#define Y 15
#define G 20


using namespace std;

int randint(int min, int max) {
  return rand() % (max + 1 - min) + min;
}

void generateWorkTime(int count, int *workers) {
  for (int i = 0; i < count; i++) {
    workers[i] = randint(0, 31);
  }
}

void generatePayment(int count, int *workers) {
  for (int i = 0; i < count; i++) {
    workers[i] = randint(10000, 15000) / 100 * 80;
  }
}

void makeSomethingElse() {
  int *yWorkTime = new int[Y];
  int *gWorkTime = new int[G];

  int *yPayment = new int[Y];
  int *gPayment = new int[G];

  generateWorkTime(Y, yWorkTime);
  generateWorkTime(G, gWorkTime);

  generatePayment(Y, yPayment);
  generatePayment(G, gPayment);

  cout << "Work time Y: " << endl;
  for (int i = 0; i < Y; i++) {
    cout << "> " << yWorkTime[i] << " ";
  }
  cout << endl;

  cout << "Work time G: " << endl;
  for (int i = 0; i < G; i++) {
    cout << "> " << gWorkTime[i] << " ";
  }
  cout << endl;

  cout << "Work payment Y: " << endl;
  for (int i = 0; i < Y; i++) {
    cout << yPayment[i] << " - ";
  }
  cout << endl;

  cout << "Work payment G: " << endl;
  for (int i = 0; i < G; i++) {
    cout << gPayment[i] << " - ";
  }
  cout << endl;

  // calculate middle payment
  int middlePaymentY;
  int middlePaymentG;

  for (int i = 0; i < G; i++) {
    middlePaymentY += yPayment[i];
  }
  middlePaymentY = middlePaymentY / Y;

  for (int i = 0; i < G; i++) {
    middlePaymentG += gPayment[i];
  }
  middlePaymentG = middlePaymentG / G;

  if (middlePaymentY > middlePaymentG) {
    cout << "Y had more averaga payment than G" << endl;
  }
  else {
    cout << "G had more averaga payment than Y" << endl;
  }

}

int main() {
  srand(time(NULL));

  makeSomethingElse();

  return 0;
}
